### PR TITLE
Cleanup ObjectManager usage - Magento_Captcha

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This PR cleanups ObjectManager usage in Magento_Captcha, by removing object manager instantiation from non-api classes

### Fixed Issues (if relevant)

N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Captcha/Controller/Refresh/Index.php [hunk 1]
```diff
@@ -1,42 +1,46 @@
 <?php
```
Comment (lbajsarowicz) [inline app/code/Magento/Captcha/Controller/Refresh/Index.php]: We are going to have conflicts in this file, as I changed this also for my PR #27381

File: app/code/Magento/Captcha/Model/Customer/Plugin/AjaxLogin.php [hunk 2]
```diff
@@ -16,70 +21,70 @@
...
-     * @throws \RuntimeException
+     * @param Json $serializer
      */
     public function __construct(
         CaptchaHelper $helper,
         SessionManagerInterface $sessionManager,
         JsonFactory $resultJsonFactory,
         array $formIds,
-        \Magento\Framework\Serialize\Serializer\Json $serializer = null
+        Json $serializer
```
Comment (lbajsarowicz) [inline app/code/Magento/Captcha/Model/Customer/Plugin/AjaxLogin.php]: Could you alias that to `JsonSerializer`? It's unclear when only `Json` is used if it's `Result\Json` or `Serialize\Json` or other `Json`-type element

File: app/code/Magento/Captcha/Model/Customer/Plugin/AjaxLogin.php [hunk 3]
```diff
@@ -16,70 +21,70 @@
...
     /**
      * Check captcha data on login action.
      *
-     * @param \Magento\Customer\Controller\Ajax\Login $subject
-     * @param \Closure $proceed
-     * @return $this
+     * @param Login $subject
+     * @param callable $proceed
+     *
+     * @return AjaxLogin|\Magento\Framework\Controller\Result\Json
```
Comment (lbajsarowicz) [inline app/code/Magento/Captcha/Model/Customer/Plugin/AjaxLogin.php]: Import as `JsonResult` alias

File: app/code/Magento/Captcha/Model/Customer/Plugin/AjaxLogin.php [hunk 4]
```diff
@@ -16,70 +21,70 @@
...
+     *
      * @SuppressWarnings(PHPMD.NPathComplexity)
      * @SuppressWarnings(PHPMD.CyclomaticComplexity)
      */
     public function aroundExecute(
-        \Magento\Customer\Controller\Ajax\Login $subject,
-        \Closure $proceed
+        Login $subject,
+        callable $proceed
```
Comment (lbajsarowicz) [inline app/code/Magento/Captcha/Model/Customer/Plugin/AjaxLogin.php]: `\Closure`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Just minor changes to be introduced.

**Comment by Bartlomiejsz** [comment]
Done @lbajsarowicz, please review

**Comment by lbajsarowicz** [comment]
Please don't `force-push` in the future, as it's easier to review the changes from recent review.
The PRs are squashed this way or another before merge, so you don't need to worry about too many commits.

**Comment by Bartlomiejsz** [comment]
Closing for same reason as https://github.com/magento/magento2/pull/27855 - will be recreated for 2.5.0

**Pull request closed without merging**